### PR TITLE
Support mocha testing of react components using enzyme

### PR DESCRIPTION
**Pull request by zacklitzsinger**
This changeset adds support for mocha/enzyme for testing purposes. It also swaps out the existing `pre-commit` library in favor of `husky` to allow for more control over git hooks. This allows us to run tests before pushes. Note that a separate changeset will alter how our linting works slightly to be more efficient and possibly be integrated with webpack.

### CODE REVIEW

File: test/sass_compiler.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Allow sass to be compiled and loaded in mocha
+
+var hook = require('css-modules-require-hook');
+var sass = require('node-sass');
+
+hook({
+    extensions: ['.scss'],
+    preprocessCss: function (css) {
```
Comment (dennismphil) [inline test/sass_compiler.js:8]: nit: This function takes `scss` as input. Should we rename the argument to `scss` instead of `css`

**Review by dennismphil — APPROVED** [review]

File: test/login.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import {expect} from 'chai';
+import {shallow} from 'enzyme';
+import {LoginBase} from '../src/Login';
+import theme from '../src/Login/style';
+
+describe('<Login />', () => {
+
+    it('should include username input', () => {
```
Comment (jessemao) [inline test/login.js:9]: should we provide a basic validation in login form?

File: test/register.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import {expect} from 'chai';
+import {shallow} from 'enzyme';
+import {RegisterBase} from '../src/Register';
+import theme from '../src/Login/style';
+
+describe('<Register />', () => {
+
+    it('should include username input', () => {
```
Comment (jessemao) [inline test/register.js:9]: should we add some validation test for register form?

**Review by jessemao — COMMENTED** [review]
I think we should include validation test for forms.

File: test/login.js (hunk 2 above)
Comment (zacklitzsinger) [reply test/login.js:9]: What kind of validation would we want? I don't know if it's safe to make any assumptions about what usernames should look like for example, except for maybe character length. A lot of that gets into customization which we will need to do, but I think is better served separately.

**Review by jessemao — APPROVED** [review]

**Pull request merged**